self-contained/bootstrap_install.sh

```diff
@@ -67,104 +67,136 @@ installNetworkPlugin() {
-        local shim_prefix="containerd-shim-"
-        local registry_path
-        local base_path
-        local shim_filename
-
-        # figure out version suffix, shims to download, and paths
-        if [ "$shim_version" == "v0.15.1" ]; then
-            version_suffix="-v2"
-            shims_to_download=("spin")
-            registry_path="oss/binaries/spinkube/containerd-shim-spin"
-            base_path="spinkube"
-            shim_filename="containerd-shim-spin-v2"
-        else
-            version_suffix="-v1"
-            shims_to_download=("spin" "slight")
-            registry_path="oss/binaries/deislabs/containerd-wasm-shims"
-            base_path="containerd-wasm-shims"
-            shim_filename="containerd-wasm-shims-linux-${CPU_ARCH}.tar.gz"
-            if [ "$shim_version" == "v0.8.0" ]; then
-                shims_to_download+=("wws")
-            fi
+    for version in "${package_versions[@]}"; do
+        local shims_to_download=("spin" "slight")
+        if [[ "$version" == "0.8.0" ]]; then
+            shims_to_download+=("wws")
+        fi
+        containerd_wasm_url=$(evalPackageDownloadURL ${PACKAGE_DOWNLOAD_URL})
+        downloadContainerdWasmShims $download_location $containerd_wasm_url "v$version" "${shims_to_download[@]}" # adding v to version for simplicity
+    done
+    wait ${WASMSHIMPIDS[@]}
+    for version in "${package_versions[@]}"; do
+        local shims_to_download=("spin" "slight")
+        if [[ "$version" == "0.8.0" ]]; then
+            shims_to_download+=("wws")
         fi
+        updateContainerdWasmShimsPermissions $download_location "v$version" "${shims_to_download[@]}"
+    done
+}
 
-        # check if shims are already downloaded
-        shims_missing=false
-        for shim in "${shims_to_download[@]}"; do
-            if [ ! -f "${containerd_wasm_filepath}/${shim_prefix}${shim}-${binary_version}${version_suffix}" ]; then
-                shims_missing=true
-                break
-            fi
-        done
+downloadContainerdWasmShims() {
+    local containerd_wasm_filepath=${1}
+    local containerd_wasm_url=${2}
+    local shim_version=${3}
+    shift 3 # Shift past the first 3 arguments to capture the rest as an array. If the max array goes larger than 3, increase accordingly.
+    local shims_to_download=("$@") # Capture the remaining arguments as an array
+    local binary_version="$(echo "${shim_version}" | tr . -)" # replaces . with - == 1.2.3 -> 1-2-3
+
+    if wasmFilesExist "$containerd_wasm_filepath" "$shim_version" "-v1" "${shims_to_download[@]}"; then
+        return
+    fi
 
-        if [ "$shims_missing" = false ]; then
-            # all shims are already downloaded, skip downloading
-            continue
-        fi
+    # Oras download for WASM for Network Isolated Clusters
+    BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER="${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER:=}"
+    if [[ ! -z ${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER} ]]; then
```

Comment:
@cameronmeissner and does bootstrap also need the oras download portion?

Reply:
yes it should, at the end of the day all provisioning logic will be the same, all the scripts themselves will simply be pre-baked onto the VHD